### PR TITLE
fix: bump terraform version from 1.3.10 to 1.5.7 

### PR DESCRIPTION
**Pull request by daniel-cit**
Bump Terrafom version from [1.3.10](https://github.com/hashicorp/terraform/tree/v1.3.10) to [1.5.7](https://github.com/hashicorp/terraform/tree/v1.5.7)

Version 1.5.7 is the last version before the License change from [Mozilla Public License, version 2.0](https://github.com/hashicorp/terraform/blob/v1.5.7/LICENSE) to [Business Source License 1.1](https://github.com/hashicorp/terraform/blob/main/LICENSE)

### CODE REVIEW

File: 0-bootstrap/README.md
Comment (apeabody) [inline 0-bootstrap/README.md:70]: Instead of specifically `1.5.7`, can/should this be the "same" version?

File: helpers/foundation-deployer/README.md
Comment (apeabody) [inline helpers/foundation-deployer/README.md:9]: I think all the go.mod are 1.22 now?

```suggestion
- [Go](https://go.dev/doc/install) 1.22 or later
```

File: helpers/foundation-deployer/README.md
Comment (apeabody) [inline helpers/foundation-deployer/README.md:53]:
```suggestion
- Check if required tools, Go 1.22.0+, Terraform 1.5.7+, gcloud 393.0.0+, and Git 2.28.0+, are installed:
```

**Review by apeabody — APPROVED** [review]

**Comment by eeaton** [comment]
Looks like all checks have passed except conventonalcommits.org has stalled, i couldn't manually retrigger it but found that editing the PR title could retrigger it. All checks are now green.

Non-blocking comment: Issue #1265 was asking for a version with security support, which only comes with 1.7 or 1.8, so I'm not sure that these changes fully address the feature request. I don't fully understand the implications of the license change that occurred after 1.5.7, but assuming we are unwilling to use versions with the new license then this fix an explanation in text is sufficient.

**Comment by ofek** [comment]
Are there plans to support newer versions of Terraform? We are concerned that using these blueprints presents a risk when we must use newer versions for bug fixes.

**Comment by apeabody** [comment]
> Are there plans to support newer versions of Terraform? We are concerned that using these blueprints presents a risk when we must use newer versions for bug fixes.

Hi @ofek - You can use a newer version of Terraform (just needs to be consistent), the documentation on how to set can be found here: https://github.com/terraform-google-modules/terraform-example-foundation/blob/main/0-bootstrap/README-GitHub.md#requirements

**Comment by ofek** [comment]
Hey there Andrew, thanks! Yes I saw that but was curious if newer versions are actually tested here. Also I'm not completely familiar with the licensing implications; is the decision that documentation will always reference `1.5.7`?

**Comment by apeabody** [comment]
> Hey there Andrew, thanks! Yes I saw that but was curious if newer versions are actually tested here. Also I'm not completely familiar with the licensing implications; is the decision that documentation will always reference `1.5.7`?

I'll defer to the repo maintainers on the roadmap question, however given the Terraform version compatibility I would not expect issues using more recent **v1** Terraform releases.  As an example I believe this repo's linter is currently using Terraform v1.9.x.

**Pull request merged**
